### ISSUE TITLE
state: manage with redux or with a graphql client ?

### ISSUE THREAD

**Issue by abenhamdine** [Arhia/arhia-next-ui#19]
This is VERY important because it also has an impact on the underlying server stack if we choose a graphql API  vs a REST API.

### tl;dr

redux/redux-saga <=> REST API
*vs*
 relay (or apollo) <=> GraphQl API

the local state not related to the databse can be managed with https://github.com/apollographql/apollo-link-state

### Some materials:

redux vs relay: 
- https://stackoverflow.com/questions/38036543/difference-between-redux-and-relay
- https://www.reindex.io/blog/redux-and-relay/
- https://github.com/reactjs/redux/issues/775
- https://medium.com/@matt.krick/replacing-relay-with-redux-2990c81aa807

apollo graphql client: 
- https://www.apollographql.com/docs/react/
- https://github.com/apollographql/react-apollo
- https://dev-blog.apollodata.com/full-stack-react-graphql-tutorial-582ac8d24e3b

Apollo pros : 
- Apollo seems less 'automagic' than relay

**Comment by abenhamdine** [comment]
closed because we decided :  
- graphql for state related to db  
- react-contextual for local ui state
